### PR TITLE
Allows PATCH and PUT request to edit components

### LINKED ISSUE #3866: Allow updating components via the REST API.
**Is your feature request related to a problem? Please describe.**

I would like to be able to update some component information using the REST API

**Describe the solution you'd like**

Enable a PUT/PATCH request to `api/components/<project-name>/<component-name>/`.

For example, with the below payload I can change the slug name.

```
update_payload = {"slug": "super-cool"}
```

**Describe alternatives you've considered**

I have checked the REST API docs and there does not seem to be a valid way to do this.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/91421567-allow-updating-components-via-the-rest-api?utm_campaign=plugin&utm_content=tracker%2F253393&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F253393&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by SaptakS**
Fixes #3866 

- [x] Every commit has a message which describes it
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them
- [x] Any code changes come with test case
- [x] Any new functionality is covered by user documentation

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -977,6 +977,132 @@ Components
...
+    :>json object project: the translation project; see :http:get:`/api/projects/(string:project)/`
+    :>json string repo: VCS repository URL
+    :>json string template: base file for monolingual translations
+    :>json string new_base: base file for adding new translations
+    :>json string vcs: version control system
+    :>json string repository_url: URL to repository status; see :http:get:`/api/components/(string:project)/(string:component)/repository/`
+    :>json string translations_url: URL to translations list; see :http:get:`/api/components/(string:project)/(string:component)/translations/`
+    :>json string lock_url: URL to lock status; see :http:get:`/api/components/(string:project)/(string:component)/lock/`
+    :>json string changes_list_url: URL to changes list; see :http:get:`/api/components/(string:project)/(string:component)/changes/`
+    :>json string push: URL of a push repository
```
Comment (nijel) [inline docs/api.rst:1008]: Many of the fields are read only or links to other API endpoints, at least following do not make sense to update: license_url, repository_url, repository_url, lock_url, changes_list_url

Same applies to PATCH below.

Comment (SaptakS) [reply docs/api.rst:1008]: These are the outputs, not the input. :thinking:

Comment (nijel) [reply docs/api.rst:1008]: Ah, I didn't realize that. Maybe it would be useful to document inputs here and reference to GET for the outputs?

Comment (SaptakS) [reply docs/api.rst:1008]: Yeah. Makes sense.

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**